chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/TestStandardReviewer.java

```diff
@@ -27,29 +28,33 @@ public TestStandardReviewer(TestStandardDAO testStandardDao, ErrorMessageUtil ms
 
     @Override
     public void review(PendingCertifiedProductDTO listing) {
-        for (PendingCertificationResultDTO cert : listing.getCertificationCriterion()) {
-            if (cert.getMeetsCriteria() != null && cert.getMeetsCriteria().equals(Boolean.TRUE)) {
-                if (cert.getTestStandards() != null && cert.getTestStandards().size() > 0) {
-                    Iterator<PendingCertificationResultTestStandardDTO> testStandardIter = cert.getTestStandards().iterator();
-                    while (testStandardIter.hasNext()) {
-                        PendingCertificationResultTestStandardDTO testStandard = testStandardIter.next();
-                        if (StringUtils.isEmpty(testStandard.getName())) {
-                            listing.getErrorMessages().add(
-                                    msgUtil.getMessage("listing.criteria.missingTestStandardName",
-                                    Util.formatCriteriaNumber(cert.getCriterion())));
-                        } else {
-                            TestStandardDTO foundTestStandard =
-                                    testStandardDao.getByNumberAndEdition(testStandard.getName(), listing.getCertificationEditionId());
-                            if (foundTestStandard == null) {
-                                listing.getWarningMessages().add(
-                                        msgUtil.getMessage("listing.criteria.testStandardNotFound",
-                                        Util.formatCriteriaNumber(cert.getCriterion()),
-                                        testStandard.getName(),
-                                        listing.getCertificationEdition()));
-                            }
-                        }
-                    }
-                }
+        List<PendingCertificationResultDTO> attestedCriteriaWithTestStandards =
+                listing.getCertificationCriterion().stream()
+                    .filter(cert -> (cert.getMeetsCriteria() != null && cert.getMeetsCriteria().equals(Boolean.TRUE)))
+                    .filter(cert -> (cert.getTestStandards() != null && cert.getTestStandards().size() > 0))
+                    .collect(Collectors.<PendingCertificationResultDTO>toList());
```

Comment:
Could you combine this stream with the next? Something like

```
listing.getCC.stream
.filter()
.filter()
.foreach(crit -> crit.getTestStandards.stream()
.forEach(ts -> reviewTS()
```

Then there wouldn't even be a need for the "attestedCriteriaWithTestStandards" variable

Also, I went back and forth on the double filters, but some StackOverflow stuff suggests that there's a negligible performance difference between two filters with simple conditions, and one filter with a complex one, and that readability should be key, so am good with the double filters as written